vortex-layout/src/layouts/row_idx/mod.rs

```diff
@@ -131,12 +131,7 @@ impl LayoutReader for RowIdxLayoutReader {
         row_offset: u64,
         splits: &mut BTreeSet<u64>,
     ) -> VortexResult<()> {
-        // Since RowIdx isn't a field, we only need to register splits for the child layout
-        // if there are any fields in the mask at all.
-        if !field_mask.is_empty() {
-            self.child.register_splits(field_mask, row_offset, splits)?;
-        }
-        Ok(())
```

Comment:
can you say more why this is wrong?

Reply:
Cause we never ask the child layout for its splits